### PR TITLE
Kinto in Docker Compose

### PR DESCRIPTION
**Pull request by mythmon**
- dev: Add a kinto server to the dev docker-compose
- test: Use a real kinto server for tests

### CODE REVIEW

**Comment by mythmon** [comment]
@leplatrem, Can you take a look at this use of Kinto in our docker-compose?

File: merino-adm/src/remote_settings/client.rs [hunk 1]
```diff
@@ -482,27 +497,30 @@ mod tests {
         ChangesetResponse, RemoteSettingsAttachmentsCapability, RemoteSettingsCapabilities,
         RemoteSettingsClient, RemoteSettingsServerInfo,
     };
-    use merino_settings::providers::RemoteSettingsConfig;
+    use lazy_static::lazy_static;
     use reqwest::Url;
 
+    lazy_static! {
```
Comment (pjenvey) [inline merino-adm/src/remote_settings/client.rs:503]: pretty nitpicky here but static strs would do here without the need for `lazy_static`, especially with `format!` accepting them anyway (and `clone()` basically the same as `str::to_string/owned`)

`static BUCKET: &str = "main";`

File: merino-integration-tests/src/utils/test_tools.rs [hunk 2]
```diff
@@ -67,25 +68,37 @@ where
...
-    });
-    settings.remote_settings.server = remote_settings_mock.base_url();
+    // Set up Remote Settings
+    let reqwest_client = reqwest::Client::new();
+    let bucket_info: serde_json::Value = reqwest_client
+        .post(format!("{}/v1/buckets/", settings.remote_settings.server))
+        .send()
+        .await
+        .expect("creating RS bucket")
+        .json()
```
Comment (pjenvey) [inline merino-integration-tests/src/utils/test_tools.rs:78]: always nice to have
```suggestion
        .error_for_status()
        .expect("Error creating bucket")
        .json()
```

File: merino-adm/src/remote_settings/client.rs (hunk 1 above)
Comment (mythmon) [reply merino-adm/src/remote_settings/client.rs:503]: Thanks for pointing this out. I don't remember why I even did this. I remember it seemed weird at the time too.

**Comment by mythmon** [comment]
Squashed and rebased to get rid of all the intermediate CI commits.

File: merino-integration-tests/src/utils/test_tools.rs [hunk 3]
```diff
@@ -67,25 +68,39 @@ where
...
-                        "base_url": remote_settings_mock.base_url()
-                    }
-                }
-            }));
-    });
-    settings.remote_settings.server = remote_settings_mock.base_url();
+    // Set up Remote Settings
+    let reqwest_client = reqwest::Client::new();
+    let bucket_info: serde_json::Value = reqwest_client
+        .post(format!("{}/v1/buckets/", settings.remote_settings.server))
```
Comment (leplatrem) [inline merino-integration-tests/src/utils/test_tools.rs:74]: Unless you run multiple concurrent tests, you could also do a `.put(format!("{}/v1/buckets/{}", settings.remote_settings.server, settings.remote_settings.bucket_id()))`

File: dev/kinto.ini [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+kinto.storage_backend = kinto.core.storage.memory
+kinto.storage_url =
+kinto.cache_backend = kinto.core.cache.memory
+kinto.cache_url =
+kinto.permission_backend = kinto.core.permission.memory
+kinto.permission_url =
+
+kinto.bucket_create_principals = system.Everyone
+kinto.bucket_read_principals = system.Everyone
+kinto.bucket_write_principals = system.Everyone
```
Comment (leplatrem) [inline dev/kinto.ini:14]: Nice :) I don't think I ever tried that! I always configure authentication :smile:

**Review by leplatrem — APPROVED** [review]
I didn't see anything wrong ;)

File: merino-integration-tests/src/utils/test_tools.rs (hunk 3 above)
Comment (mythmon) [reply merino-integration-tests/src/utils/test_tools.rs:74]: We run multiple concurrent tests.

**Review by pjenvey — APPROVED** [review]

**Pull request merged**